### PR TITLE
Add examples of System.Private.CoreLib iterative workflow

### PR DESCRIPTION
**Pull request by safern**
cc: @danmosemsft @stephentoub

### CODE REVIEW

**Comment by safern** [comment]
Oops, missed to update build.sh, will update that.

File: docs/workflow/building/libraries/README.md [hunk 1]
```diff
@@ -176,6 +176,17 @@ dotnet build System.Net.NetworkInformation.csproj /p:TargetOS=Linux
 dotnet build -c Release System.Net.NetworkInformation.csproj
 ```
 
+### Iterating on System.Private.CoreLib changes
+When changing `System.Private.CoreLib` after a full build, in order to test against those changes, you will need an updated `System.Private.CoreLib` in the testhost. In order to achieve that, you can build the `libs.pretest` subset which does testhost setup including copying over `System.Private.CoreLib`.
+
+```
+build.cmd clr.corelib+clr.nativecorelib+libs.pretest -rc Release
```
Comment (danmosemsft) [inline docs/workflow/building/libraries/README.md:183]: I still have to build the runtime first, right? Maybe this should make clear that you would likely do a command like XYZ first, then repeat the command here as you iterate on S.P.C.

File: eng/build.ps1 [hunk 2]
```diff
@@ -93,9 +93,12 @@ function Get-Help() {
   Write-Host "* Build Mono runtime for Windows x64 on Release configuration."
   Write-Host ".\build.cmd mono -c release"
   Write-Host ""
-  Write-Host "It's important to mention that to build Mono for the first time,"
-  Write-Host "you need to build the CLR and Libs subsets beforehand."
-  Write-Host "This is done automatically if a full build is performed at first."
+  Write-Host "* Build Release coreclr corelib, crossgen corelib and update Debug libraries testhost to run test on an updated corelib."
+  Write-Host ".\build.cmd clr.corelib+clr.nativecorelib+libs.pretest -rc release"
+  Write-Host ""
+  Write-Host "* Build Release mono corelib and update Debug libraries testhost to run test on an updated corelib."
```
Comment (danmosemsft) [inline eng/build.ps1:99]: Release mono corelib? Looks like debug?

File: eng/build.sh [hunk 3]
```diff
@@ -107,9 +107,12 @@ usage()
   echo "* Build Mono runtime for Linux x64 on Release configuration."
   echo "./build.sh mono -c release"
   echo ""
-  echo "It's important to mention that to build Mono for the first time,"
-  echo "you need to build the CLR and Libs subsets beforehand."
-  echo "This is done automatically if a full build is performed at first."
+  echo "* Build Release coreclr corelib, crossgen corelib and update Debug libraries testhost to run test on an updated corelib."
+  echo "./build.sh clr.corelib+clr.nativecorelib+libs.pretest -rc release"
+  echo ""
+  echo "* Build Release mono corelib and update Debug libraries testhost to run test on an updated corelib."
```
Comment (danmosemsft) [inline eng/build.sh:113]: same comment about debug/release

**Review by danmoseley — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]
Other than Dan's comments, LGTM. Thanks.

**Comment by ViktorHofer** [comment]
FWIW we should start binplacing CoreLib (and optionally crossgen it) so that these manual steps aren't necessary. cc @ericstj

**Review by ViktorHofer — APPROVED** [review]

**Comment by safern** [comment]
> FWIW we should start binplacing CoreLib (and optionally crossgen it) so that these manual steps aren't necessary

We should be careful when doing that because if CoreLib is built via a P2P and the configuration for CoreLib is Debug, but the runtime was built in release, there will be a configuration miss match in between CoreLib and runtime and I believe that is not allowed.

File: docs/workflow/building/libraries/README.md (hunk 1 above)
Comment (safern) [reply docs/workflow/building/libraries/README.md:183]: Yeah, you should have done `build.cmd clr` or `build.cmd clr.runtime` first. I'll add a comment to clarify that.

**Comment by ViktorHofer** [comment]
> configuration miss match in between CoreLib and runtime and I believe that is not allowed.

@jkotas is mixing configurations between the runtime and CoreLib problematic? I assume not...

**Comment by jkotas** [comment]
> @jkotas is mixing configurations between the runtime and CoreLib problematic? 

Yes, it is problematic. Layout of some data structures embedded in CoreLib is different between debug and release.

**Pull request merged**